### PR TITLE
bisect: elements of a are bounded by SupportsRichComparison

### PR DESCRIPTION
**Pull request by Kentzo**

### CODE REVIEW

File: stdlib/_bisect.pyi [hunk 1]
```diff
@@ -1,47 +1,28 @@
 import sys
-from _typeshed import SupportsRichComparison, SupportsRichComparisonT
+from _typeshed import SupportsRichComparison
 from typing import Callable, MutableSequence, Sequence, TypeVar, overload
 
-_T = TypeVar("_T")
+_T = TypeVar("_T", bound=SupportsRichComparison)
```
Comment (JelleZijlstra) [inline stdlib/_bisect.pyi:5]: Conventionally T means an unbounded TypeVar. Can you rename this to `_ SupportsRichComparisonT`?

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Requesting changes for my previous comment.

File: stdlib/_bisect.pyi (hunk 1 above)
Comment (Kentzo) [reply stdlib/_bisect.pyi]: @JelleZijlstra Perhaps it would be better to import and re-use `SupportsRichComparisonT`?

**Comment by Kentzo** [comment]
@JelleZijlstra up

File: stdlib/_bisect.pyi [hunk 2]
```diff
@@ -1,51 +1,75 @@
...
+        lo: int = ...,
+        hi: int | None = ...,
+        *,
+        key: None = ...,
+    ) -> int: ...
     @overload
     def bisect_left(
-        a: Sequence[_T],
-        x: SupportsRichComparisonT,
+        a: Sequence[_SupportsRichComparisonT],
```
Comment (JelleZijlstra) [inline stdlib/_bisect.pyi:21]: Why do we still need a bounded TypeVar in this overload? If there's a key function, we don't need to support comparisons on the list elements themselves, right?

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Sorry for the delay!

**Pull request merged**